query/src/sql/planner/binder/ddl/table.rs

```diff
@@ -38,31 +42,150 @@ use crate::sql::ColumnBinding;
 use crate::sql::ScalarExpr;
 use crate::sql::OPT_KEY_DATABASE_ID;
 
+struct SelectBuilder {
+    from: String,
+    columns: Vec<String>,
+    filters: Vec<String>,
+    order_bys: Vec<String>,
+}
+
+impl SelectBuilder {
+    fn from(table_name: &str) -> SelectBuilder {
+        SelectBuilder {
+            from: table_name.to_owned(),
+            columns: vec![],
+            filters: vec![],
+            order_bys: vec![],
+        }
+    }
+    fn with_column(&mut self, col_name: impl Into<String>) -> &mut Self {
+        self.columns.push(col_name.into());
+        self
+    }
+
+    fn with_filter(&mut self, col_name: impl Into<String>) -> &mut Self {
+        self.filters.push(col_name.into());
+        self
+    }
+
+    fn with_order_by(&mut self, order_by: &str) -> &mut Self {
+        self.order_bys.push(order_by.to_owned());
+        self
+    }
+
+    fn build(self) -> String {
+        let columns = {
+            let s = self.columns.join(",");
+            if !s.is_empty() {
+                s
+            } else {
+                "*".to_owned()
+            }
+        };
+
+        let order_bys = {
+            let s = self.order_bys.join(",");
+            if !s.is_empty() {
+                format!("ORDER BY {s}")
+            } else {
+                s
+            }
+        };
+
+        let filters = {
+            let s = self.filters.join(" and ");
+            if !s.is_empty() {
+                format!("where {s}")
+            } else {
+                "".to_owned()
+            }
+        };
+
+        let from = self.from;
+        format!("SELECT {columns} FROM {from} {filters} {order_bys} ")
+    }
+}
+
 impl<'a> Binder {
     pub(in crate::sql::planner::binder) async fn bind_show_tables(
         &mut self,
+        bind_context: &BindContext,
         stmt: &ShowTablesStmt<'a>,
     ) -> Result<Plan> {
         let ShowTablesStmt {
-            database,
+            database: db,
             full,
             limit,
             with_history,
         } = stmt;
 
-        let database = database.as_ref().map(|ident| ident.name.to_lowercase());
-        let kind = match limit {
-            Some(ShowLimit::Like { pattern }) => PlanShowKind::Like(pattern.clone()),
-            Some(ShowLimit::Where { selection }) => PlanShowKind::Like(selection.to_string()),
-            None => PlanShowKind::All,
+        let mut database = self.ctx.get_current_database();
+        if let Some(ident) = db {
+            database = ident.name.to_lowercase()
+        }
+
+        if DatabaseCatalog::is_case_insensitive_db(&database) {
+            database = database.to_uppercase()
```

Comment:
Why case insensitive db use uppercase while the sensitive db use lowercase?

Reply:
Because in map the db_name is upper. But I see that in query/src/catalogs/default/database_catalog.rs the API in DatabaseCatalog has already judge is_case_insensitive_db. So we do not need to judge it on the parser.